core/arch/emit_utils_shared.c

```diff
@@ -1897,12 +1903,13 @@ append_jmp_to_fcache_target(dcontext_t *dcontext, instrlist_t *ilist,
         if (shared) {
             /* next_tag placed into tls slot earlier in this routine */
 #ifdef AARCH64
-            /* Load next_tag from FCACHE_ENTER_TARGET_SLOT (TLS_REG2_SLOT):
-             * ldr x0, [x28]
+            /* Load next_tag from FCACHE_ENTER_TARGET_SLOT, stored by
+             * append_setup_fcache_target.
              */
             APP(ilist,
-                XINST_CREATE_load(dcontext, opnd_create_reg(DR_REG_X0),
-                                  OPND_CREATE_MEMPTR(dr_reg_stolen, 16)));
+                XINST_CREATE_load(
+                    dcontext, opnd_create_reg(DR_REG_X0),
+                    OPND_CREATE_MEMPTR(dr_reg_stolen, FCACHE_ENTER_TARGET_SLOT)));
```

Comment:
This could be instr_create_restore_from_tls()?

Reply:
Done.